test/unit/math/opencl/device_functions/digamma_test.cpp

```diff
@@ -0,0 +1,32 @@
+#ifdef STAN_OPENCL
+#include <stan/math.hpp>
+#include <stan/math/opencl/device_functions/digamma.hpp>
+#include <stan/math/opencl/kernel_cl.hpp>
+#include <gtest/gtest.h>
+#include <test/unit/math/expect_near_rel.hpp>
+#include <string>
+
+static const std::string test_kernel_code
+    = STRINGIFY(__kernel void test(__global double *B, __global double *A) {
+        const int i = get_global_id(0);
+        B[i] = digamma(A[i]);
+      });
+
+const stan::math::opencl_kernels::kernel_cl<
+    stan::math::opencl_kernels::out_buffer,
+    stan::math::opencl_kernels::in_buffer>
+    digamma("test", {stan::math::opencl_kernels::digamma_device_function,
+                     test_kernel_code});
+
+TEST(MathMatrixCL, digamma) {
+  Eigen::VectorXd a = Eigen::VectorXd::Random(1000).array() * 30;
```

Comment:
Are there any edge cases with digamma in boost math that we should test for?

Reply:
No idea. Maybe some mathematician should decide?

Reply:
I meant more in the sense does the calculation change for some interval of x?
If not I would still like to see some similar test to the once in scal/digamma
https://github.com/stan-dev/math/blob/develop/test/unit/math/prim/scal/fun/digamma_test.cpp

To see if we replicate the NaN stuff and large arguments.

Reply:
I searched for something like this test, but i missed it. I will add those tests.